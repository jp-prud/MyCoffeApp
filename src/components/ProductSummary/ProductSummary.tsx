import {Image, Pressable} from 'react-native';

import {ProductSummaryProps} from '@domain';
import {useNavigation} from '@react-navigation/native';
import {formatPrice} from '@utils';

import {Text, Box} from '@components';

export function ProductSummary({product}: {product: ProductSummaryProps}) {
  const navigate = useNavigation();

  if (!product) {
    return;
  }

  function handleClickNavigateProductPage() {
    navigate.navigate('ProductScreen', {
      productId: product.id,
    });
  }

  return (
    <Pressable
      onPress={handleClickNavigateProductPage}
      testID="product-summary">
      <Box width={154} borderRadius="s16" overflow="hidden" elevation={1}>
        <Box
          height={154}
          borderRadius="s12"
          overflow="hidden"
          justifyContent="center"
          alignItems="center">
          <Image
            source={{
              uri: product?.images[0].url,
              height: 152,
              width: 152,
            }}
            style={{borderRadius: 14}}
            alt={product?.images[0].alt}
          />
        </Box>

        <Box p="s12" gap="s8">
          <Text preset="paragraphSmall" numberOfLines={1}>
            {product.name}
          </Text>

          <Text preset="paragraphSmall" bold>
            {formatPrice(product.price)}
          </Text>
        </Box>
      </Box>
    </Pressable>
  );
}
