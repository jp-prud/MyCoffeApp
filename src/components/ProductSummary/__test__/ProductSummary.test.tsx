import React from 'react';

import {ProductSummaryProps} from '@domain';

import {render} from '../../../../test/test-utils';
import {ProductSummary} from '../ProductSummary';

describe('ProductSummary', () => {
  describe('when ProductSummary passed', () => {
    const productMock: ProductSummaryProps = {
      id: '1',
      name: 'Product 1',
      price: 100,
      images: [
        {
          url: 'https://via.placeholder.com/150',
          alt: 'Product 1',
        },
      ],
    };

    // eslint-disable-next-line jest/expect-expect
    it('should render ProductSummary', () => {
      render(<ProductSummary product={productMock} />);
    });
  });

  describe('when ProductSummary not passed', () => {
    const productMock: ProductSummaryProps = {
      id: '1',
      name: 'Product 1',
      price: 100,
      images: [
        {
          url: 'https://via.placeholder.com/150',
          alt: 'Product 1',
        },
      ],
    };

    // eslint-disable-next-line jest/expect-expect
    it('should render ProductSummary', () => {
      render(<ProductSummary product={productMock} />);
    });
  });
});
