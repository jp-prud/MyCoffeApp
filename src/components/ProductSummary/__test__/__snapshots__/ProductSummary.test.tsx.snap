// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSummary ProductData was passed should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="product-summary"
>
  <View
    style={
      [
        {
          "borderRadius": 16,
          "elevation": 1,
          "overflow": "hidden",
          "width": 154,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 12,
            "overflow": "hidden",
            "width": 154,
          },
        ]
      }
    >
      <Image
        alt="Product 1"
        source={
          {
            "height": 152,
            "uri": "https://via.placeholder.com/150",
            "width": 154,
          }
        }
      />
    </View>
    <View
      style={
        [
          {
            "gap": 8,
            "padding": 12,
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "#101010",
            },
            [
              {
                "fontSize": 14,
                "lineHeight": 19.6,
              },
              {
                "fontFamily": "Inter-Regular",
              },
              undefined,
            ],
          ]
        }
      >
        Product 1
      </Text>
      <Text
        style={
          [
            {
              "color": "#101010",
            },
            [
              {
                "fontSize": 14,
                "lineHeight": 19.6,
              },
              {
                "fontFamily": "Inter-Bold",
              },
              undefined,
            ],
          ]
        }
      >
        R$Â 100,00
      </Text>
    </View>
  </View>
</View>
`;
